VisualPinball.Unity/VisualPinball.Unity/Mappings/MappingConfig.cs

```diff
@@ -139,23 +168,23 @@ private static ISwitchDeviceComponent GuessSwitchDevice(ISwitchDeviceComponent[]
 
 				var swMatch = switchDevices.FirstOrDefault(sw => sw.name == matchKey);
 				if (swMatch != null) {
-					return swMatch;
+					switches.Add(swMatch);
 				}
 			}
+			else if (!string.IsNullOrEmpty(engineSwitch.DeviceHint)) {
+				foreach (var device in switchDevices) {
+					var regex = new Regex(engineSwitch.DeviceHint, RegexOptions.IgnoreCase);

```

Comment:
If you use the static instance of Regex instead of a new instance of it each time you can get a performance increase of .NET's built in RegEx caching. Additionally, if this is doing far more "lookups" then setting it to compiled is also a good idea as you'll get a perf hit the first call, but subsequent calls will be faster as it will be precompiled and cached.

So like this:
```cs 
Regex.IsMatch(device.name, engineSwitch.DeviceHint, RegexOptions.IgnoreCase | RegexOptions.Compiled)
```

See here: https://docs.microsoft.com/en-us/dotnet/standard/base-types/compilation-and-reuse-in-regular-expressions

Reply:
There are a few other places as well, here it doesn't make much of a difference because it's a one-time thing and editor-only, but it might be worth optimizing elsewhere too.

Reply:
Ohh didn’t realize this is editor only.